### PR TITLE
Adding Splunk Returner

### PR DESCRIPTION
**Pull request by scottjpack**
Splunk Returner using HTTP Event Collector

Requires some pillar/config changes

### CODE REVIEW

File: salt/returners/splunk.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
+# -*- coding: utf-8 -*-
+'''
+
+Send json response data to Splunk via the HTTP Event Collector
+Requires the following config values to be specified in config or pillar:
+
```
Comment (jfindlay) [inline salt/returners/splunk.py]: @scottjpack, `.. versionadded:: Boron`.

File: salt/returners/splunk.py [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    payload.update({"event":event})
+    logging.info("Payload: %s" % json.dumps(payload))
+     
+         #fire it off
+    splunk_event.sendEvent(payload)
+    return True
+
+
+
+# Thanks to George Starcher for the http_event_collector class (https://github.com/georgestarcher/)
```
Comment (cachedout) [inline salt/returners/splunk.py]: Is there a license for this code? If so, could you please direct me to it?

**Comment by cachedout** [comment]
Hi @scottjpack Thanks for sending this in. I left a comment in-line but didn't hear back from you. Your comment suggests that some of this code came from another repo and I just need to know precisely where so we can ensure that we're following the proper licensing. We can't merge this until that's done. Thanks!

**Comment by cachedout** [comment]
@georgestarcher Hi there. We got a PR that gives attribution to using some of your code. Could you take a look here and please let me know if it's all right if we merge this into Salt? Thanks.

**Comment by scottjpack** [comment]
Hey Mike,

George releases under the MIT license.  From his repo description:

Version/Date: 1.0.0 2015-11-27

Author: George Starcher (starcher) Email: george@georgestarcher.com

This code is presented *AS IS* under MIT license.

**Comment by georgestarcher** [comment]
Yup. Have at it. 


-------
George Starcher @georgestarcher
http://www.georgestarcher.com/

GnuPG key ID: 875A3320BD558C9E
Fingerprint: 0555 5F20 3615 CC61 67B9 1409 875A 3320 BD55 8C9E

**Comment by cachedout** [comment]
Hi @scottjpack Thanks for the reply. There are a number of lint failures here: https://jenkins.saltstack.com/job/salt-pr-lint-n/12862/violations/file/salt/returners/splunk.py/

Could you please make the suggested changes and push a commit up? Then we'll have a good look at things and see if we can get this in. Thanks!

**Comment by cachedout** [comment]
@georgestarcher Thank you very much for getting back to me! Much appreciated.

**Comment by cachedout** [comment]
@scottjpack Looks like some of those lint errors are fixed up now but a few remain. Let me know if you have questions on any of them. Thanks!

**Comment by cachedout** [comment]
Merged via #30907

**Pull request closed without merging**